teams/core.toml

```diff
@@ -50,38 +50,6 @@ address = "user-logos@rust-lang.org"
 [[lists]]
 address = "legal@crates.io"
```

Comment:
We might want to move this away from core too? Dunno if it's ever been used. Otherwise looks good to me.

Note for onlookers: still used address are not being deleted, but they're now managed outside the team repository.

Reply:
It's not something I'm sure about either but I'd prefer to not touch it just yet.